src/ASM/ASMs2D.C

```diff
@@ -3362,26 +3436,29 @@ void ASMs2D::BasisFunctionCache::calculateAll ()
                            std::vector<BasisFunctionVals>& values)
   {
     std::vector<Go::BasisDerivsSf>  spline;
-    patch.getBasis(basis)->computeBasisGrid(par1,par2,spline);
+    const Go::SplineSurface* surf = basis == 0 ? patch.getGeometry() : patch.getBasis(basis);
+    surf->computeBasisGrid(par1,par2,spline);
     size_t idx = 0;
     for (const Go::BasisDerivsSf& spl : spline) {
       SplineUtils::extractBasis(spl,values[idx].N,values[idx].dNdu);
       ++idx;
     }
   };
-  if (nderiv == 1)
+  if (nderiv == 1) {
     extract1(mainQ->gpar[0],mainQ->gpar[1],values);
-  else if (nderiv == 2) {
+  } else if (nderiv == 2) {
     std::vector<Go::BasisDerivsSf2> spline;
-    patch.getBasis(basis)->computeBasisGrid(mainQ->gpar[0],mainQ->gpar[1],spline);
+    const Go::SplineSurface* surf = basis == 0 ? patch.getGeometry() : patch.getBasis(basis);
+    surf->computeBasisGrid(mainQ->gpar[0],mainQ->gpar[1],spline);
```

Comment:
The local scope variable surf here may conflict with member with same name (at least on newer compilers)?
An alternative could be to redefine the `getBasis()` method to return the value of `getGeometry()` when the argument is zero, if that always holds. Then the change here would not be needed at all.